### DISCUSSION TITLE
Issue of using non-uniform mesh in THM

### DISCUSSION THREAD

**Discussion by SomnusYu** [idaholab/moose#27435] [Q&A Modules: Thermal Hydraulics] [answered]
Hi all,

I would like to use non-uniform mesh in the pipe component. The input file is shown below:
[pipe_with_non_uniform_mesh.txt](https://github.com/idaholab/moose/files/15057181/pipe_with_non_uniform_mesh.txt)

I find there's an abrupt change in pressure at the junction. 
![image](https://github.com/idaholab/moose/assets/118275988/55c94d9d-6897-48eb-9df5-dcff7cc2aa4c)

How to fix this problem? Thanks!

**Comment by GiudGiud** [comment]
Hello

Some junctions will have a pressure loss. It s expected and it's physical. Mixing flows, changing directions etc come at a cost

@joshuahansel will know whether this is the case for this input

Guillaume

**Comment by joshuahansel** [comment]
If you just want to have varying element sizes along a pipe, you don't need junctions. You can specify multiple values in the "length" and "n_elems" parameters for each of your subsections. Please try that to see if you still get the jump. Nothing in your input file suggests there should be a jump there (physically there should be no loss there). If you still get the jump, that suggests it related to the transition of the element sizes, not the presence of the junction component. In that case, I suspect the slope reconstruction to be at fault. Try setting "rdg_slope_reconstruction = none" and see if the jump persists. These aren't solutions, just troubleshooting.

**Reply by SomnusYu** [reply]
Hi @joshuahansel,

I try to use "n_elems" and "length" instead of the junction component. Also, I try to use "rdg_slope_reconstruction = none". 
[pipe_with_non_uniform_mesh.txt](https://github.com/idaholab/moose/files/15070025/pipe_with_non_uniform_mesh.txt)

But this problem still exists.

**Reply by SomnusYu** [reply]
I think the problem is we use flux to calculate the connection point rather than the nodal value. So we must refine the mesh as much as possible. The question is if we can use the non-refined mesh to do this. Using the volumeJunction component to offset this impact? I'm not sure.

**Reply by joshuahansel** [reply, marked as answer]
I tried "rdg_slope_reconstruction = FULL", and it appears to resolve the issue. This means that the jump is an artifact of spatial discretization error. We've seen this kind of behavior before.

**Reply by SomnusYu** [reply]
Thanks! “rdg_slope_reconstruction = FULL” works very well!

**Reply by SomnusYu** [reply]
Hi @joshuahansel,

Sometimes I need to use other slop methods (FULL may not converge), what should I do in this case?

**Reply by joshuahansel** [reply]
You can try the other methods. If nothing else works, I think your only option is just to refine your mesh with `NONE`.

**Reply by SomnusYu** [reply]
Thanks!
